include/cuco/extent.cuh

```diff
@@ -32,8 +33,8 @@ static constexpr std::size_t dynamic_extent = static_cast<std::size_t>(-1);
  * @tparam N Extent
  */
 template <typename SizeType, std::size_t N = dynamic_extent>
-struct extent {
-  using value_type = SizeType;  ///< Extent value type
+struct extent : private detail::extent_base<SizeType> {
+  using value_type = typename detail::extent_base<SizeType>::value_type;  ///< Extent value type
```

Comment:
Adding this base class was also necessary so I could provide a trait which checks if a given type is `cuco::extent`-like